mathics/eval/parts.py

```diff
@@ -179,8 +180,11 @@ def walk_levels(
                 cur_pos + [index + 1],
             )
             depth = max(element_depth + 1, depth)
-            elements.append(element)
-        new_expr = make_expression(head, *elements)
+            elements.append(new_element)
+            if new_element is not element:
+                elem_prop = None
+
+        new_expr = make_expression(head, *elements, elements_properties=elem_prop)
```

Comment:
Here is where the element_properties were lost. Now, for large blocks, I see a ~5% of improvement in performance in the folliwing operations:
```
In[1]:= a = Table[1./(1+i^2),{i,1000}];Timing[Plus@@a]
Out[1]= {0.00908414, 1.07567}

In[2]:= Timing[MatchQ[a, {__Real}]]
Out[2]= {0.00175423, True}

In[3]:= F[x__Real]:=1; Timing[F@@a]
Out[3]= {0.00755716, 1}
```